src/clients/SpokePoolClient.ts

```diff
@@ -453,9 +464,23 @@ export class SpokePoolClient extends BaseAbstractClient {
       }
     }
 
+    const toBlock = this.eventSearchConfig.toBlock || (await this.spokePool.provider.getBlockNumber());
+
+    // Determine if this spoke pool has the capability to bridge UDSC via the CCTP token bridge.
+    // The CCTP bridge is canonically disabled if the `cctpTokenMessenger` is the ZERO address.
+    let hasCCTPBridgingEnabled = false;
+    if (chainIsCCTPEnabled(this.chainId)) {
```

Comment:
Could this also be conditional on `this.chainId !== hubPoolClient.chainId` ? That'd save the lookup on mainnet.

Reply:
Fair point - but `chainIsCCTPEnabled(1) === false` on the function definition. This should be false for mainnet and would not need the additional lookup